examples/r2i/classification.cc

```diff
@@ -0,0 +1,333 @@
+    auto params = factory->MakeParameters (error);
+    error = params->Configure(engine, model);
+    if (error.IsError ()) {
+      std::cerr << "Params configuration error: " << error << std::endl;
+      exit(EXIT_FAILURE);
+    }
+    params->Set ("input-layer", in_node);
+    params->Set ("output-layer", out_node);
+  }
+
+  std::cout << "Loading image: " << image_path << std::endl;
+  std::unique_ptr<unsigned char[]> image_data = LoadImage (image_path, pre_width,
+      pre_height, pre_channels);
+
+  std::cout << "Configuring input frame" << std::endl;
+  std::shared_ptr<r2i::IFrame> in_frame = factory->MakeFrame (error);
+
+  error = in_frame->Configure (image_data.get(), pre_width, pre_height,
+                               pre_format_id);
+  if (error.IsError ()) {
+    std::cerr << "Output frame configuration error: " << error << std::endl;
+    exit(EXIT_FAILURE);
+  }
+
+  std::cout << "Configuring output frame" << std::endl;
+  std::shared_ptr<r2i::IFrame> out_frame = factory->MakeFrame (error);
+  std::shared_ptr<float> out_data = std::shared_ptr<float>
+                                    (new float[pre_width * pre_height * pre_channels],
+                                     std::default_delete<float[]>());
+
+  error = out_frame->Configure (out_data.get(), pre_width, pre_height,
+                                pre_format_id);
+  if (error.IsError ()) {
+    std::cerr << "Input frame configuration error: " << error << std::endl;
+    exit(EXIT_FAILURE);
+  }
+
+  std::cout << "Applying pre processing" << std::endl;
+  error = preprocessing->Apply (in_frame, out_frame);
+  if (error.IsError ()) {
+    std::cerr << "Preprocessing error: " << error.GetDescription() << std::endl;
+    exit(EXIT_FAILURE);
+  }
+
+  std::cout << "Starting engine" << std::endl;
+  error = engine->Start ();
+  if (error.IsError ()) {
+    std::cerr << "Engine start error: " << error << std::endl;
+    exit(EXIT_FAILURE);
+  }
+
+  std::cout << "Predicting..." << std::endl;
+  std::vector< std::shared_ptr<r2i::IPrediction> > predictions;
+  error = engine->Predict (out_frame, predictions);
+
+  std::cout << "Apply post processing" << std::endl;
+  std::vector< std::shared_ptr<r2i::InferenceOutput> > ioutputs;
+  std::shared_ptr<r2i::IPostprocessing> postprocessing =
+    loader->LoadPostprocessing(
+      postprocess_path, error);
+  if (error.IsError ()) {
+    std::cerr << "Postprocessing loading error: " << error << std::endl;
+    exit(EXIT_FAILURE);
+  }
+
+  error = postprocessing->Apply(predictions, ioutputs);
+  if (error.IsError ()) {
+    std::cerr << "Postprocessing error: " << error << std::endl;
+    exit(EXIT_FAILURE);
+  }
+
+  for (uint i = 0 ; i < ioutputs.size(); i++) {
+    auto ioutput = ioutputs.at(i);
+    std::cout << "  Processing output: " << i << std::endl;
+
+    if (ioutput->GetType() != r2i::CLASSIFICATION) {
+      std::cerr <<
+                "Wrong Inference Output type, this output is not classification type" <<
```

Comment:
I think this would read better as: "Wrong inference output type, this model is not for classification, or an erroneous post-processing was used."